src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -79,4 +104,13 @@ public void removeConnection(SourceRemovedEvent e) {
             }
         }
     }
+
+    /**
+     * Thrown when the user tries to create a connection that will result in an infinite loop.
+     */
+    public static class InfiniteLoopException extends Error {
```

Comment:
I'd rather use exceptions over expecting future developers to call another validation method prior to using the connection class. This really is a unique error that would cause a legitimate concern.